webapp/tests/test_render_glyph.py

```diff
@@ -0,0 +1,1002 @@
+          glyph.find_x_times(None, glyph.SEC, 1.0)
+
+      with self.assertRaises(ValueError):
+          glyph.find_x_times(self.dt, None, 1.0)
+
+      with self.assertRaises(ValueError):
+          glyph.find_x_times(self.dt, glyph.SEC, None)
+
+      with self.assertRaises(ValueError):
+          glyph.find_x_times('', glyph.SEC, 1.0)
+
+      with self.assertRaises(ValueError):
+          glyph.find_x_times(self.dt, '', 1.0)
+
+      with self.assertRaises(ValueError):
+          glyph.find_x_times(self.dt, glyph.SEC, '')
+
+      with self.assertRaises(ValueError):
+          glyph.find_x_times(self.dt, glyph.YEAR, 1)
+
+      with self.assertRaises(ValueError):
+          glyph.find_x_times(self.dt, glyph.SEC, 0)
+
+      with self.assertRaises(ValueError):
+          glyph.find_x_times(self.dt, glyph.SEC, 1.0)
+
+
+class LinearAxisTicsTest(TestCase):
+
+    def test_LinearAxisTics_valid_input(self):
+      self.assertTrue(glyph._LinearAxisTics(0.0, 100.0, unitSystem='binary'))
+
+    def test_LinearAxisTics_invalid_min_value(self):
+      with self.assertRaises(glyph.GraphError):
+          glyph._LinearAxisTics(float('NaN'), 100.0, unitSystem='binary')
+
+    def test_LinearAxisTics_invalid_max_value(self):
+      with self.assertRaises(glyph.GraphError):
+          glyph._LinearAxisTics(0.0, float('inf'), unitSystem='binary')
+
+    #
+    # Testing _LinearAxisTics.checkFinite()
+    #
+
+    def test_LinearAxisTics_checkFinite_nan(self):
+      y = glyph._LinearAxisTics(0.0, 100.0, unitSystem='binary')
+      with self.assertRaises(glyph.GraphError):
+          y.checkFinite(float('nan'))
+
+    def test_LinearAxisTics_checkFinite_inf(self):
+      y = glyph._LinearAxisTics(0.0, 100.0, unitSystem='binary')
+      with self.assertRaises(glyph.GraphError):
+          y.checkFinite(float('inf'))
+
+    def test_LinearAxisTics_checkFinite_100(self):
+      y = glyph._LinearAxisTics(0.0, 100.0, unitSystem='binary')
+      self.assertEqual(y.checkFinite(100), 100)
+
+    #
+    # Testing _LinearAxisTics.chooseDelta()
+    #
+
+    def test_LinearAxisTics_chooseDelta_0_1(self):
+      y = glyph._LinearAxisTics(0.0, 100.0, unitSystem='binary')
+      self.assertEqual(y.chooseDelta(1), 0.1)
+
+    def test_LinearAxisTics_chooseDelta_very_small(self):
+      y = glyph._LinearAxisTics(0.0, 100.0, unitSystem='binary')
+      self.assertEqual(y.chooseDelta(1.0e-10), 1.0)
+
+    #
+    # Testing _LinearAxisTics.applySettings()
+    #
+
+    def test_LinearAxisTics_applySettings_axisLimit_min_greater_max(self):
+      y = glyph._LinearAxisTics(0.0, 100.0, unitSystem='binary')
+      with self.assertRaises(glyph.GraphError):
+          y.applySettings(axisMin=1000, axisMax=10, axisLimit=None)
+
+    #
+    # Testing _LinearAxisTics.reconcileLimits()
+    #
+
+    def test_LinearAxisTics_reconcileLimits_defaults(self):
+      y = glyph._LinearAxisTics(0.0, 100.0, unitSystem='binary')
+      self.assertEqual(y.reconcileLimits(), None)
```

Comment:
The `AxisTics` classes were meant to be used by constructing an object and then calling the instance's `applySettings()`, `chooseStep()`, and `chooseLimits()` methods in sequence. Why? Because that was the smallest change from the previous code.

`reconcileLimits()` is really meant to be called only by `applySettings()`, and only in the case that `not (self.minValue < self.maxValue)`. So I think it is inappropriate for the tests in this section to be calling `reconcileLimits()` directly.

OTOH I think it would make sense to move the `not (self.minValue < self.maxValue)` check into `reconcileLimits()` to make it a little bit more self-contained. We could even make the method private, though when I was coding this I think I checked and didn't find many private methods, so I thought they were disfavored in this codebase.

I'm working on some patches to clear up minor points like this.